core/src/test/java/org/apache/iceberg/rest/TestRESTCatalog.java

```diff
@@ -2347,7 +2347,7 @@ public void testPaginationForListNamespaces() {
     RESTCatalog catalog =
         new RESTCatalog(SessionCatalog.SessionContext.createEmpty(), (config) -> adapter);
     catalog.initialize("test", ImmutableMap.of(RESTSessionCatalog.REST_PAGE_SIZE, "10"));
-    int numberOfItems = 30;
+    int numberOfItems = 27;
```

Comment:
Just curious why you are modifying the number of items here? Is there a specific case you are testing?

Reply:
This test case only works if `numberOfItems % pageSize == 0`, where `pageSize` is hardcoded to "10" in the line right above. Anything else will show the error in the original paging code. That's why I first changed the test, then fixed the code.

Reply:
I'm resolving this thread since the discussion is the same as above. Please do reopen if you think I resolved this prematurely.